can-event-queue.js

```diff
@@ -25,36 +25,53 @@ var ensureMeta = function ensureMeta(obj) {
 // getHandlers returns a KeyTree used for event handling.
 // `handlers` will be on the `can.meta` symbol on the object.
 function getHandlers(obj) {
-    var meta = ensureMeta(obj);
-
-    var handlers = meta.handlers;
-    if(!handlers) {
-        // Handlers are organized by:
-        // event name - the type of event bound to
-        // binding type - "event" for things that expect an event object (legacy), "onKeyValue" for reflective bindings.
-        // queue name - mutate, queue, etc
-        // handlers - the handlers.
-        handlers = meta.handlers = new KeyTree([Object, Object, Object, Array],{
-            onFirst: function(){
-                if( obj._eventSetup ) {
-                    obj._eventSetup();
-                }
-            },
-            onEmpty: function(){
-                if( obj._eventTeardown ) {
-                    obj._eventTeardown();
-                }
-            }
-        });
-    }
-    return handlers;
+	var meta = ensureMeta(obj);
+
+	var handlers = meta.handlers;
+	if(!handlers) {
+		// Handlers are organized by:
+		// event name - the type of event bound to
+		// binding type - "event" for things that expect an event object (legacy), "onKeyValue" for reflective bindings.
+		// queue name - mutate, queue, etc
+		// handlers - the handlers.
+		handlers = meta.handlers = new KeyTree([Object, Object, Object, Array],{
+			onFirst: function(){
+				if( obj._eventSetup ) {
+					obj._eventSetup();
+				}
+				dispatchOnKeyTree(obj, getMetaHandlers(obj), "boundChange", [true]);
+			},
+			onEmpty: function(){
+				if( obj._eventTeardown ) {
+					obj._eventTeardown();
+				}
+				dispatchOnKeyTree(obj, getMetaHandlers(obj), "boundChange", [false]);
+			}
+		});
+	}
+	return handlers;
 }
 
-// These are the properties we are going to add to objects
-var props = {
-	dispatch: function(event, args){
+// getMetaHandlers returns a KeyTree used for handling internal events (like handling first binding).
+// `metaHandlers` will be on the `can.meta` symbol on the object.
+function getMetaHandlers(obj) {
```

Comment:
I would call this `lifecycleHandlers` ... meta handlers is too confusing with `can.meta`.

Reply:
Another thing that might help, is to make `ensureMeta` builds both `handlers` and `lifecycleHandlers` at the same time.  I think this will improve code clarity and likely performance.